packages/esm-service-queues-app/src/queue-screen/queue-screen.component.tsx

```diff
@@ -1,15 +1,69 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { DataTableSkeleton } from '@carbon/react';
 import { useTranslation } from 'react-i18next';
 import { useActiveTickets } from './useActiveTickets';
 import PatientQueueHeader from '../patient-queue-header/patient-queue-header.component';
 import styles from './queue-screen.scss';
+import { useSelectedQueueLocationUuid } from '../helpers/helpers';
 
 interface QueueScreenProps {}
 
 const QueueScreen: React.FC<QueueScreenProps> = () => {
   const { t } = useTranslation();
-  const { activeTickets, isLoading, error } = useActiveTickets();
+  const { activeTickets, isLoading, error, mutate } = useActiveTickets();
+  const speaker = window.speechSynthesis;
+  const [isSpeaking, setIsSpeaking] = useState(false);
+  const selectedLocation = useSelectedQueueLocationUuid();
+  const locationFilteredTickets = activeTickets;
+
+  const rowData = locationFilteredTickets.map((ticket, index) => ({
+    id: `${index}`,
+    room: ticket.room,
+    ticketNumber: ticket.ticketNumber,
+    status: ticket.status,
+  }));
+
+  function readTicket(queue, speaker) {
+    if ('speechSynthesis' in window) {
+      const message = new SpeechSynthesisUtterance();
+      const utterance =
+        'Ticket Number: ' +
+        queue.ticketNumber.split('-')[0].split('') +
+        ', - ' +
+        queue.ticketNumber.split('-')[1].split('') +
+        ', Please Proceed To Room, ' +
+        queue.room;
+      message.rate = 1;
+      message.pitch = 1;
+      message.text = utterance;
+      return new Promise((resolve) => {
+        message.onend = resolve;
+        speaker.speak(message);
+      });
+    } else {
+      return Promise.resolve();
+    }
+  }
```

Comment:
This could be moved into the component function body and memoized with a `useCallback`:

```tsx
  const readTicket = useCallback((queue) => {
    if ('speechSynthesis' in window) {
      const message = new SpeechSynthesisUtterance();
      const [prefix, suffix] = queue.ticketNumber.split('-');
      const utterance = `Ticket Number: ${prefix.split('')}, - ${suffix.split('')}, Please Proceed To Room, ${queue.room}`;
      message.rate = 1;
      message.pitch = 1;
      message.text = utterance;
      return new Promise<void>((resolve) => {
        message.onend = resolve;
        speaker.speak(message);
      });
    }
    return Promise.resolve();
  }, [speaker]);
```

Also, if the `SpeechSynthesisUtterance` API is locale-aware (it probably is), we should consider internationalising `utterance` using something like:

```tsx
	 const utterance = t(
          'ticketAnnouncement',
          'Ticket number: {{prefix}}, - {{suffix}}, please proceed to room {{room}}',
          {
            prefix: prefix.split('').join(' '),
            suffix: suffix.split('').join(' '),
            room: queue.room,
          },
        );
```